server/cluster/cluster.go

```diff
@@ -327,6 +320,15 @@ func (c *RaftCluster) Start(s Server) error {
 		log.Error("load external timestamp meets error", zap.Error(err))
 	}
 
+	// bootstrap keyspace group manager after starting other parts successfully.
```

Comment:
Why change the order?

Reply:
This and `m.ctx.Done()` are a double insurance measure.

Reply:
If the `keyspaceGroupManager` has started a goroutine according to the previous sequence, when the raft cluster creation fails, it will return without running `s.stopRaftCluster()` https://github.com/tikv/pd/issues/7001#issuecomment-1701550466

Reply:
Shall we cancel it?

Reply:
Do you think to check `m.ctx.Done()` is enough?

Reply:
I mean we need to use a separate context to handle the case you mentioned.

Reply:
I got it, but I don't think it's necessary. The goroutines below don't use another context either, but since they're at the tail of the order, they're not stuck.